src/Voron/StorageEnvironment.cs

```diff
@@ -554,7 +540,23 @@ internal void TransactionCompleted(LowLevelTransaction tx)
             if (tx.Flags != (TransactionFlags.ReadWrite))
                 return;
 
+            if (tx.FlushedToJournal)
+            {
+                var totalPages = 0;
+                // ReSharper disable once LoopCanBeConvertedToQuery
+                foreach (var page in tx.GetTransactionPages())
+                {
+                    totalPages += page.NumberOfPages;
+                }
+
+                Interlocked.Add(ref SizeOfUnflushedTransactionsInJournalFile, totalPages);
+
+                if (tx.IsLazyTransaction == false)
```

Comment:
It shouldn't matter. Unless, you have noticed another potential race condition here, have you? The key was that it had to be called after `ActiveTransactions.TryRemove(tx)`.